### PR TITLE
he2019momentum_2+zayne-sprague: Initial Draft (no images yet)

### PR DESCRIPTION
**Pull request by Zayne-sprague**

### CODE REVIEW

File: _summaries/he2019momentum_2.md [hunk 1]
```diff
@@ -3,17 +3,100 @@ layout: summary
...
-* What interesting variants are explored?
+## What's the big idea
+
+A large dynamic dictionary of keys in a queue, built via a momentum encoder, can improve contrastive learning making unsupervised pretraining in computer vision viable and better than supervised pretraining.
+
+## What's hard about unsupervised pre-training in vision
+
+- Pretraining on images is difficult because they are in a continuous space, and the representations learned may not transfer
+
+- Some work has been done on pretraining computer vision models using pretext tasks, but by in large have failed to outperform supervised pretraining
```
Comment (patelojas) [inline _summaries/he2019momentum_2.md:18]: The paper gives a good comparison between NLP tasks and CV for pretraining and representation. I think including that comparison might help readers understand the context of the problem at hand more.

File: _summaries/he2019momentum_2.md [hunk 2]
```diff
@@ -3,17 +3,100 @@ layout: summary
...
+- Given some input $$q$$ and a list of keys $$k_i$$ we want to have a low value for when $$q$$ and $$k_+$$ have similar representation and have a high value when they are different.
+
+- One way to measure the similarity between queries and keys is by the dot product.
+
+- We can then measure the basic loss of a contrastive algorithm by:
+
+$$L_q = - log \frac{exp(q \cdot k_+ / t)}{\sum^K_{t=0}exp(q \cdot k_i / t)}$$
+
+- where $$q$$ is the current image query, $$k_+$$ is the correct key that corresponds to the query, $$k_i$$ are negative samples, and $$t$$ is a temperature hyperparameter.
+
```
Comment (patelojas) [inline _summaries/he2019momentum_2.md:39]: Good summary of the calculations

File: _summaries/he2019momentum_2.md [hunk 3]
```diff
@@ -3,17 +3,100 @@ layout: summary
...
+<p align="center">
+  <img width="1200" alt="Pretext Tasks" src="./he2019momentum_2d.png">
+</p>
+
+- MoCo also outperforms every model on nearly every down stream task (note the one failure being semantic segmentations on VOC)
+
+<p align="center">
+  <img width="1200" alt="Downstream Tasks" src="./he2019momentum_2c.png">
+</p>
+
```
Comment (patelojas) [inline _summaries/he2019momentum_2.md:82]: I think there are more results and tables you can look to include, such as on VOC object detection.

File: _summaries/he2019momentum_2.md [hunk 4]
```diff
@@ -3,17 +3,100 @@ layout: summary
...
 ## TL;DR
-* Three
-* Bullets
-* To highlight the core concepts
+
+* Contrastive Loss benefits from a large queue-like dictionary of negative samples, where sufficiently old activations are dropped
+
+* Contrastive Loss benefits from momentum weights on the key encoder to produce better keys and thus help the query encoder train better representations
+
+* MoCo and Pretext tasks are now better at pretraining computer vision models than supervised alternatives
```
Comment (patelojas) [inline _summaries/he2019momentum_2.md:102]: Good summary, you explain things in an easy to understand manner. Left a few comments on things that might be useful to include.

File: _summaries/he2019momentum_2.md (hunk 1 above)
Comment (Zayne-sprague) [reply _summaries/he2019momentum_2.md:18]: Good point, I added a bit more to this section, mostly introducing why its easy in NLP vs why its hard in CV

File: _summaries/he2019momentum_2.md (hunk 3 above)
Comment (Zayne-sprague) [reply _summaries/he2019momentum_2.md:95]: Oh right, I added that result table.

File: _summaries/he2019momentum_2.md (hunk 4 above)
Comment (Zayne-sprague) [reply _summaries/he2019momentum_2.md:115]: Thanks!

**Review by zhaoyue-zephyrus — APPROVED** [review]
